AxisRotationConstraint.cs

```diff
@@ -0,0 +1,138 @@
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.SpatialManipulation;
+using UnityEngine;
+
+public class AxisRotationConstraint : TransformConstraint
+{
+    #region Properties
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around X-Axis is possible.")]
+    private bool allowRotationOnXAxis;
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around Y-Axis is possible.")]
+    private bool allowRotationOnYAxis;
+
+    [SerializeField]
+    [Tooltip("Set to true, if rotation around Z-Axis is possible.")]
+    private bool allowRotationOnZAxis;

```

Comment:
Why are movementOn*Axis serialized fields? You force this value to `true` in `Awake`.   It looks like you are using this variable to track some private state, and it shouldn't be serialized.